src/Menu/ControlledMenuItem.js

```diff
@@ -111,71 +111,77 @@ class MenuItem extends React.PureComponent {
             </Flex>
           )}
         </Flex>
-      </StyledMenuItem>
+      </MenuItemComponent>
     );
   }
 }
 
-MenuItem.propTypes = {
+ControlledMenuItem.propTypes = {
   /**
-   * Indicates whether or not the menu item is selected
+   * Base color to use for the menu item
    */
-  selected: PropTypes.bool,
+  baseColor: PropTypes.string,
 
   /**
-   * Callback to run when a selection is made. This is generally intended for
-   * internal use with the Menu component.
+   * Name of icon to use for the menu item
    */
-  onSelect: PropTypes.func,
+  iconName: PropTypes.string,
 
   /**
-   * Main label text for the menu item
+   * HTML ID for the menu item
    */
-  label: PropTypes.string.isRequired,
+  id: PropTypes.string.isRequired,
 
   /**
-   * Secondary label text for the menu item
+   * Whether or not the current menu item is focused
    */
-  secondaryLabel: PropTypes.string,
+  isFocused: PropTypes.bool,
 
   /**
-   * HTML input name property for the input field
+   * Indicates whether or not the menu item is selected
    */
-  name: PropTypes.string.isRequired,
+  isSelected: PropTypes.bool,
 
   /**
-   * HTML input value property for the input field
+   * Main label text for the menu item
    */
-  value: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
 
   /**
-   * Base color to use for the menu item
+   * Component used to render each menu item
    */
-  baseColor: PropTypes.string,
+  MenuItemComponent: PropTypes.elementType.isRequired,
```

Comment:
Since your providing a default via `defaultProps` for this, should this no longer be required?

Reply:
Good call